mypy/checkexpr.py

```diff
@@ -789,17 +789,21 @@ def check_typeddict_call_with_kwargs(
         context: Context,
         orig_callee: Type | None,
     ) -> Type:
-        if not (callee.required_keys <= set(kwargs.keys()) <= set(callee.items.keys())):
+        actual_keys = kwargs.keys()
+        if not (callee.required_keys <= actual_keys <= callee.items.keys()):
             expected_keys = [
                 key
                 for key in callee.items.keys()
-                if key in callee.required_keys or key in kwargs.keys()
+                if key in callee.required_keys or key in actual_keys
             ]
-            actual_keys = kwargs.keys()
             self.msg.unexpected_typeddict_keys(
                 callee, expected_keys=expected_keys, actual_keys=list(actual_keys), context=context
             )
-            return AnyType(TypeOfAny.from_error)
+            if callee.required_keys > actual_keys:
```

Comment:
This condition is unreachable, there is `callee.required_keys <= actual_keys` in the enclosing if statement.

Reply:
Hello!

I believe this is correct as the `callee.required_keys <= actual_keys` has a not preceeding it.

Indeed, removing this return wil fail the test `[case TestCannotCreateTypedDictInstanceWithMissingItems]`


Let's imagine the following:

```
assert callee.items.keys == {1,2,3,4}
assert callee.required_keys == {1,2,3}
actual_keys = {1,2}

if not (required <= actual_keys <= callee.items.keys ):
    # call unexpected_typeddict_keys on the actual vs expected

    if required > actual:
        return AnyType
```

Will return AnyType and print out a single error message: "missing required key 3"

Whereas with `actual_keys = {1,2,3, 'unrelated'}` will print out "extra key 'unrelated' but _not_ return AnyType; ie it will continue to check the TypedDict for any other errors (For example, the actual keys might all be accounted for but have the wrong type).

Reply:
Oh yeah, right, I have missed the `not`.